src/Microsoft.ML.TimeSeries/SequentialTransformerBase.cs

```diff
@@ -407,4 +524,320 @@ public ValueGetter<TValue> GetGetter<TValue>(int col)
             }
         }
     }
+
+    /// <summary>
+    /// This class is a transform that can add any number of output columns, that depend on any number of input columns.
+    /// It does so with the help of an <see cref="IRowMapper"/>, that is given a schema in its constructor, and has methods
+    /// to get the dependencies on input columns and the getters for the output columns, given an active set of output columns.
+    /// </summary>
+
+    public sealed class TimeSeriesRowToRowMapperTransform : RowToRowTransformBase, IRowToRowMapper,
```

Comment:
>TimeSeriesRowToRowMapperTransform [](start = 24, length = 33)

Why did you need to clone it, and why is it public?

Reply:
i'll refactor this out and make it internal

---
In reply to: [235554740](https://github.com/dotnet/machinelearning/pull/1618#discussion_r235554740) [](ancestors = 235554740)